### PR TITLE
Clean up C.UTF-8 locale changes

### PR DESCRIPTION
**Pull request by tristan957**
Removes some unnecessary initdb arguments, and fixes Neon for MacOS since it doesn't seem to ship a C.UTF-8 locale.

### CODE REVIEW

**Review by arpad-m — APPROVED** [review]

**Review by erikgrinaker — APPROVED** [review]
LGTM thanks!

**Comment by skyzh** [comment]
I'll cherry-pick to release branch

**Review by mtyazici — APPROVED** [review]

File: compute_tools/src/config.rs [hunk 1]
```diff
@@ -74,10 +74,17 @@ pub fn write_postgres_conf(
     }
 
     // Locales
-    writeln!(file, "lc_messages='C.UTF-8'")?;
-    writeln!(file, "lc_monetary='C.UTF-8'")?;
-    writeln!(file, "lc_time='C.UTF-8'")?;
-    writeln!(file, "lc_numeric='C.UTF-8'")?;
+    if cfg!(target_os = "macos") {
```
Comment (clipperhouse) [inline compute_tools/src/config.rs:77]: Nit, should “macos” be a string? Is there a const or enum that would be a little safer?

File: pageserver/src/tenant.rs [hunk 2]
```diff
@@ -4786,12 +4786,6 @@ async fn run_initdb(
         .args(["--username", &conf.superuser])
         .args(["--encoding", "utf8"])
         .args(["--locale", &conf.locale])
-        .args(["--lc-collate", &conf.locale])
```
Comment (clipperhouse) [inline pageserver/src/tenant.rs:4789]: Presume we’re remove the ability to pass options for this? That hard-coding is preferred?

**Review by clipperhouse — APPROVED** [review]

File: compute_tools/src/config.rs (hunk 1 above)
Comment (skyzh) [reply compute_tools/src/config.rs:77]: yes, unfortunately, that's how Rust works now

Comment (tristan957) [reply compute_tools/src/config.rs:77]: I'm not aware of any Rust enum for cfg checks. I'll look around.

File: pageserver/src/tenant.rs (hunk 2 above)
Comment (tristan957) [reply pageserver/src/tenant.rs:4789]: Sorry, passing `--locale` also sets these options. This was feedback in the previous PR that somehow didn't make it in even though I had committed the suggestion.

File: compute_tools/src/config.rs (hunk 1 above)
Comment (tristan957) [reply compute_tools/src/config.rs:77]: Apparently this exists in a crate, but it isn't worth adding.

**Pull request merged**